File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/DamageEmblem.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class DamageEmblem extends Equipable {
	{
		name = Locale.get("damage_emblem");
		description = Locale.get("damage_emblem_desc");
		sprite = "item-more_damage";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);

		if (this.owner instanceof Player) {
			((Player) this.owner).damageModifier += 0.4f;
		}
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);

		if (this.owner instanceof Player) {
			((Player) this.owner).damageModifier -= 0.4f;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/DamageEmblem.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equipable

import org.rexcellentgames.burningknight.assets.Locale

class DamageEmblem : Equipable() {
	private val mod: Float
		get() = (20 + this.level * 20).toFloat()

	init {
		name = Locale.get("damage_emblem")
		description = Locale.get("damage_emblem_desc")
		sprite = "item-more_damage"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.damageModifier += mod / 100f
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.damageModifier -= mod / 100f
	}

	override fun getMaxLevel(): Int {
		return 4
	}

	override fun getDescription(): String {
		return super.getDescription().replace("{PERCENT}", mod.toString())
	}
}